tools/@aws-cdk/node-bundle/README.md

```diff
@@ -0,0 +1,133 @@
+# node-bundle
+
+Create bundled packages with zero dependencies and appropriate license attributions.
+
+## Why
+
+When shipping nodejs applications, there is currently no easy way to ensure your users are
+consuming the exact dependency clojure your package was tested against.
+
+This is because many libraries define their dependencies with a range, rather then a fixed version.
+NPM has provided an install time lock file called [shrinkwrap](https://docs.npmjs.com/cli/v8/commands/npm-shrinkwrap)
+to mitigate this, however, this file is only respected by NPM itself, and not by other package managers such as Yarn.
+
+## What
+
+This package wires up several popular tools to offer a simple entrypoint for
+creating self-contained nodejs packages.
+
+The resulting packages are still npm installable packages, but the entrypoints you specify are
+replaced with a bundled version of them, embedding all their dependencies inline.
+Note that embedding dependencies means you are effectively redistributing third-party software.
+This has legal implications, since it requires proper attribution
+of those dependencies, validating their licenses allow such redistribution.
+
+This tool takes care of all that:
+
+- Bundle dependencies inside the package.
+
+  > This is currently done with [esbuild](), but is subject to change.
+
+- Validate and create NOTICE files with complete third-party attributions.
+
+  > This is currently done with [license-checker](https://www.npmjs.com/package/license-checker), but is subject to change.
+
+- Enforce no circular imports are exhibited in your package, nor in your dependency clojure.
+
+  > This is currently done with [madge](https://www.npmjs.com/package/madge), but is subject to change.
+
+## Alternative Approaches
+
+We considered two other alternatives before eventually going down this route:
+
+### Bundled Dependencies
+
+Aside from a shrinkwrap file, NPM also offers a feature called `bundledDependencies`
+to vendor in your dependencies inside the `node_modules` directory of your package.
+
+> See [bundledDependencies](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#bundleddependencies)
+
+While this approach seems to be supported across all package managers, that won't be
+the case for Yarn 2.x and above, or more concretely,
+for the [Plug'n'Play](https://yarnpkg.com/features/pnp) feature.
+
+> See [dont use bundled dependencies](https://yarnpkg.com/getting-started/migration#dont-use-bundledependencies)
+
+### Static Binaries
+
+Another option would have been to produce platform specific static binaries that embed both
+dependencies as well as a node runtime.
+
+This approach is valid, but really depends on the use case. For example if you need your package
+to still be installable by npm, it doesn't really fit. Also, it's not relevant for libraries,
+only CLI applications.
+
+## How
```

Comment:
Readme is  missing a section on how/where to put stuff in your `package.json`s "scripts" section.

Also, it's not clear to me what `fix` does :)

Reply:
Oh now I see.

Isn't `bundle validate --fix` a more logical invocation there?

Reply:
> Readme is missing a section on how/where to put stuff in your package.jsons "scripts" section.

Good call 👍 

> Isn't bundle validate --fix a more logical invocation there?

Yes, much better, i'll change.